deployment/supervisor.py

```diff
@@ -102,6 +104,12 @@ def _build_workers(self, config):
         api_worker = api.ApiWorker(self.config_path, config, self.notifier, websocket_notifier, provider.authentificator(), self._health)
         workers.append(api_worker)
 
+        if config.has_option("general", "check_releases_frequency"):
```

Comment:
You can avoid this condition by using a default value
`frequency = config.getint("general", "check_releases_frequency", -1)`